### PR TITLE
QA-256: Collect and publish acceptance tests coverage

### PR DESCRIPTION
**Pull request by lluiscampos**
Namely:
* New TestMain to parse flags and call actual main (inspired in backend
  services design).
* New make target to compile the binary with coverage instrumentation.
* Test wrapper to use the instrumented binary on cli calls, creating
  unique coverage filenames.
* Extend pipeline to collect and publish the coverage stats.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,6 +39,10 @@ build-multiplatform:
 	CGO_ENABLED=0 GOOS=darwin GOARCH=amd64 $(GO) build $(GO_LDFLAGS) $(BUILDV) $(BUILDTAGS) \
 	     -o mender-cli.darwin.amd64
 
+build-coverage:
+	CGO_ENABLED=0 $(GO) test -c -o mender-cli-test \
+		-coverpkg $(shell $(GO) list ./... | grep -v vendor | grep -v mocks | grep -v test | tr  '\n' ,)
```
Comment (oleorhagen) [inline Makefile:44]: Nit:
`go list ./...` never lists the vendor dir

File: main_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/mendersoftware/mender-cli/cmd"
+)
+
+var runAcceptanceTests bool
+
+// used for parsing '-cli-args' for urfave/cli when running acceptance tests
+// this is because of a conflict between urfave/cli and regular go flags required for testing (can't mix the two)
+var cliArgsRaw string
+
+var _ = func() bool {
```
Comment (oleorhagen) [inline main_test.go:31]: Question:
Why does it need to return anything?

File: main_test.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+	// parse '-cli-args', remember about binary name at idx 0
+	var cliArgs []string
+
+	if cliArgsRaw != "" {
+		cliArgs = []string{os.Args[0]}
+		splitArgs := strings.Split(cliArgsRaw, " ")
+		cliArgs = append(cliArgs, splitArgs...)
+	}
+
+	// copy(os.Args, cliArgs)
```
Comment (oleorhagen) [inline main_test.go:56]: Leftover

**Review by oleorhagen — APPROVED** [review]
Holy cow - solid!

Been going to town here the last couple of days!

File: Makefile (hunk 1 above)
Comment (lluiscampos) [reply Makefile:44]: Good to know. It is probably wrong everywhere ([inventory](https://github.com/mendersoftware/inventory/blob/master/Dockerfile.acceptance-testing#L4-L5) as an example)

File: main_test.go (hunk 2 above)
Comment (lluiscampos) [reply main_test.go:31]: I have no idea ([reference](https://github.com/mendersoftware/inventory/blob/master/main_test.go#L30-L33)). I'll ask to backend folks.

Comment (lluiscampos) [reply main_test.go]: I didn't get response. I removed it and still works... so :man_shrugging:

Comment (oleorhagen) [reply main_test.go]: Working code we like!

**Pull request merged**